src/wallet/walletdb.cpp

```diff
@@ -59,6 +59,7 @@ const std::string WALLETDESCRIPTORCKEY{"walletdescriptorckey"};
 const std::string WALLETDESCRIPTORKEY{"walletdescriptorkey"};
 const std::string WATCHMETA{"watchmeta"};
 const std::string WATCHS{"watchs"};
+const std::unordered_set<std::string> LEGACY_TYPES{CRYPTED_KEY, CSCRIPT, DEFAULTKEY, HDCHAIN, KEYMETA, KEY, OLD_KEY, POOL, WATCHMETA, WATCHS};
```

Comment:
c50135edb6414fa5fd598f65e13eaa0e639573b9: maybe add comment: `// These keys are deleted by the migratewallet RPC` to ensure we don't add more stuff to this list without making sure they get migrated.

Note to other reviewers (plus some open questions), these types are captured by the backup as follows:
* `CRYPTED_KEY`: when `walletdb.cpp` loads them, it puts them into `mapCryptedKeys` via `LoadCryptedKey()`. `MigrateToDescriptor()` iterates over that map.
* `CSCRIPT`: when `walletdb.cpp` loads them, it puts them into `mapScripts` via `LoadCScript()`. 
   * `MigrateToDescriptor()` iterates over any multisig scripts in that map
   * TODO: I'm unclear what happens to `mapScripts` entries that are not multisig (see comment on earlier commit)
   * caveat: old records with `redeemScript.size() > MAX_SCRIPT_ELEMENT_SIZE` are lost in the upgrade (they're in the backup)
* `DEFAULTKEY`: this is only used to detect corruption, and is not migrated, newer wallets don't have this key
* `HDCHAIN`: `m_hd_chain` which we migrate
* `KEYMETA`: ends up in `mapKeyMetadata` via `LoadKeyMetadata`, which we migrate (at least for every key we have).
* `KEY`:  when `walletdb.cpp` loads them, it puts them into `mapKeys` via `LoadKey()`. `MigrateToDescriptor()` iterates over that map.
* `OLD_KEY`: these can only be loaded with v0.18 and older. They are skipped during wallet load, so they are lost in the migration.
* `POOL`: these are ignored, we regenerate the keypool using descriptors
* `WATCHMETA`: put in `m_script_metadata` by  `LoadScriptMetadata`, and indexed by their `CScriptID`, TODO: same question as with `CSCRIPT`
* `WATCHS` put in `mapWatchKeys`   via `LoadWatchOnly`, and its keys are used to monitor PKH, etc via `ImplicitlyLearnRelatedKeyScripts`. We don't iterate over `mapWatchKeys` directly, so I'm a bit confused how these are migrated.